src/datadoc_editor/frontend/callbacks/utils.py

```diff
@@ -433,3 +449,52 @@ def save_metadata_and_generate_alerts(metadata: Datadoc) -> list:
         variables_control(missing_obligatory_variables, metadata.variables),
         check_variable_names(metadata.variables),
     ]
+
+
+def map_selected_algorithm_to_pseudo_fields(
+    selected_algorithm: str,
+) -> list:
+    """Map a PseudonymizationAlgorithms enum value to the correct pseudonymization input list.
+
+    Examples:
+    >>> pseudo_fields = map_selected_algorithm_to_pseudo_fields("PAPIS_ALGORITHM_WITHOUT_STABLE_ID")
+    >>> len(pseudo_fields)
+    1
+    >>> pseudo_fields = map_selected_algorithm_to_pseudo_fields("PAPIS_ALGORITHM_WITH_STABLE_ID")
+    >>> len(pseudo_fields)
+    2
+    >>> pseudo_fields = map_selected_algorithm_to_pseudo_fields("CUSTOM")
+    >>> len(pseudo_fields)
+    5
+    """
+    mapping = {
+        "PAPIS_ALGORITHM_WITHOUT_STABLE_ID": PSEUDONYMIZATION_PAPIS_WITHOUT_STABLE_ID_METADATA,
+        "PAPIS_ALGORITHM_WITH_STABLE_ID": PSEUDONYMIZATION_PAPIS_WITH_STABLE_ID_METADATA,
+        "STANDARD_ALGORITM_DAPLA": PSEUDONYMIZATION_DEAD_METADATA,
+        "CUSTOM": PSEUDONYMIZATION_METADATA,
+    }
+
+    return mapping.get(selected_algorithm, [])
+
+
+def map_dropdown_to_pseudo(variable: model.Variable) -> str:
+    """Return dropdown algorithm value for a variable's pseudonymization."""
+    pseudonym_obj = variable.pseudonymization
+
+    if pseudonym_obj is None:
+        return ""
+
+    encryption_algorithm = pseudonym_obj.encryption_algorithm
+    if encryption_algorithm == PAPIS_ALGORITHM_ENCRYPTION:
+        if pseudonym_obj.stable_identifier_type == PAPIS_ALGORITHM_WITH_STABLE_ID_TYPE:
+            return str(
+                PseudonymizationAlgorithmsEnum.PAPIS_ALGORITHM_WITH_STABLE_ID.value
+            )
+        return str(
+            PseudonymizationAlgorithmsEnum.PAPIS_ALGORITHM_WITHOUT_STABLE_ID.value
+        )
+    if encryption_algorithm == STANDARD_ALGORITM_DAPLA_ENCRYPTION:
+        return str(PseudonymizationAlgorithmsEnum.STANDARD_ALGORITM_DAPLA.value)
+    if encryption_algorithm:
+        return str(PseudonymizationAlgorithmsEnum.CUSTOM.value)
+    return ""
```

Comment:
```suggestion
def map_dropdown_to_pseudo(
    variable: model.Variable,
) -> PseudonymizationAlgorithmsEnum | None:
    """Return dropdown algorithm value for a variable's pseudonymization."""
    if variable.pseudonymization:
        match variable.pseudonymization.encryption_algorithm:
            case constants.PAPIS_ALGORITHM_ENCRYPTION:
                if (
                    variable.pseudonymization.stable_identifier_type
                    == constants.PAPIS_ALGORITHM_WITH_STABLE_ID_TYPE
                ):
                    return PseudonymizationAlgorithmsEnum.PAPIS_ALGORITHM_WITH_STABLE_ID
                return PseudonymizationAlgorithmsEnum.PAPIS_ALGORITHM_WITHOUT_STABLE_ID
            case constants.STANDARD_ALGORITM_DAPLA_ENCRYPTION:
                return PseudonymizationAlgorithmsEnum.STANDARD_ALGORITM_DAPLA
            case None:
                return None
            case _:
                return PseudonymizationAlgorithmsEnum.CUSTOM

    return None
```

Could be easier to follow?